### DISCUSSION TITLE
How to upgrade to v2 potentially without SQL access?

### DISCUSSION THREAD

**Discussion by Nicktentjr** [prisma/prisma#9379] [Q&A] [answered]
I recently joined a company that uses Prisma 1.0 with a postgresql server, and one of my tasks is to upgrade us to Prisma 2. As I was going through the update instructions, after installing the CLI and trying to pull our database schema, I stumbled on a problem. The credentials, which we had been using thus far with no issue, weren't being accepted.

At the moment we're kind-of at a standstill, and so I asked my coworkers (there are two other developers besides me) for access to the SQL server. Perhaps from there I can see what credentials work, or if I need to adjust the database url in the .env file. Unfortunately it looks like neither of them know the exact way to access the SQL database. This presents a problem, as even if I were able to move on to the next steps, I will need to run SQL scripts to update the tables to account for the incompatibilities between versions.

We're going to try to get into the server, but if we're unable to, I'd like to know two things:
1. Is there a way to make the Prisma CLI communicate with the old Prisma 1 settings to re-establish connection to the server?
2. If I can't access the server in a way that lets me write and execute queries, is there a different way of handling the incompatibilities?
3. If this upgrade isn't going to work out, I need a way to revert back to my original settings. From where I am in the process (having just run init), is there a way to revert back to the earlier version of Prisma?

I can provide additional info in the morning if needed, including any error screens, file contents, etc. I just figured I'd at least get the ball rolling if anyone had any ideas.

**Comment by matthewmueller** [comment, marked as answer]
No, this is unfortunately not possible. We don't execute queries for you, so you'll need to be able to connect to the underlying DB. You'll also need that connection string to use Prisma 2. 

Do you have access to the production host? You might be able to find it in some environment/secrets configuration.

**Reply by Nicktentjr** [reply]
Thankfully we were able to figure out how to access our database and I'm now able to run the queries necessary.

One question I have, while running the queries given to fix enum lists, it gave this error:
"ERROR: column "columnname" contains null values." This is likely because the column in question has some leftover null records. I'm still on my local copy, but will this have negative effects when running these queries on our production server? We're trying to avoid any data loss while going through this upgrade, and I just want to double check.

**Reply by Nicktentjr** [reply]
Another frequent error I'm getting is casting issues for fixing enum data types:

"default for column [columnname] cannot be cast automatically to type [type]"

Are there workarounds for these errors too?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
